### ISSUE TITLE
Default to GHCi mode?

### ISSUE THREAD

**Issue by hamishmack** [leksah/leksah#163]
It is in the menu (Debug -> GHCi), on the toolbar and you also get prompted to start it if you try to use a feature that needs it (like evaluating an expression), but users can still overlook it.  Perhaps it should be the default for packages with no dependent packages in the workspace (we can fall back on `cabal build` if `cabal repl` fails to start).

**Comment by jaccokrijnen** [comment]
I agree, I was thinking the same thing. We should make sure though that changes to the cabal file (adding dependencies etc) trigger at least a restart of ghci. I don't think ':reload' handles this.
What is the problem with dependent packages in the workspace?

**Comment by hamishmack** [comment]
If you have `package-a` and `package-b` in you workspace where `package-b` depends on `package-a` then changes made to `package-a` should result in both `package-a` and `package-b` rebuilding.  This works in `cabal build` mode because we run cabal build, copy, register on `package-a` then the same on `package-b`.

I now think maybe we should drop the modal approach and allow multiple GHCi sessions (up to one per package in the workspace).

When there is a GHCi session running for `package-a` and a change is made to `package-a` we both `cabal build` and `:reload` it.

However when we build `package-b` as the result of a change in `pacakge-a` we still run `cabal build` on `package-b`, but instead of sending `:reload` we restart the GHCi session with `cabal repl` so that it picks up the changes in `package-a`.

By allowing GHCi session per package and still doing builds at the same time we can ditch the "Would you like to run GHCi" dialogs and just do it.  The output of each can go to its own log and the `cabal build` can continue to use the default log.

**Comment by jaccokrijnen** [comment]
Ah, I see. Hiding the ghci toggle sounds like a good plan, it shouldn't be the user's concern. 

So the point of running `cabal build` in the background would be the `register` part? I think it's not always desirable to have other projects rebuild when working on a dependency (by default). As an IDE user I would expect that the background building is only used for direct feedback on the current project. Perhaps we could use the "add-source dependency" mechanism to let the user indicate the need for automatic rebuilding instead?

It seems like 'cabal repl' should be restarted in these cases then:

* When "add-source dep" has changed
* When the cabal file changes (new deps might have to be installed).  Would there be other cases?
* When the package database has changed (user installed new versions)

Would there be other cases?

**Comment by ndmitchell** [comment]
I recommend defaulting to "fast and don't destroy the users package database mode (#165)". It is a much nicer default for the kind of things I'm doing.

**Comment by hamishmack** [comment]
> Would there be other cases?

* If the user selects a different active named component (i.e. lib, exe, test suite).  I think we have to keep the concept of an active component so that we know what to pass to `cabal repl`.  Currently you set in the Panes -> Workspace by double clicking on the package (to select lib) or one of the named components that are listed as children of the package.
